example/file_browser.c

```diff
@@ -345,7 +345,8 @@ media_init(struct media *media)
 static void
 file_browser_reload_directory_content(struct file_browser *browser, const char *path)
 {
-    strncpy(browser->directory, path, MAX_PATH_LEN);
+    const size_t path_len = strlen(path) + 1;
+    memmove(browser->directory, path, MIN(path_len, MAX_PATH_LEN));
```

Comment:
Would a `strcpy()` be better here? Or explicitly calling it like... `strncpy(browser->directory, path, strlen(path));`?

Reply:
Both `strcpy()` (https://en.cppreference.com/w/c/string/byte/strcpy) and `strncpy()` (https://en.cppreference.com/w/c/string/byte/strncpy) have `restrict` pointer parameters, so both would trigger undefined behavior.

Reply:
Thanks for the explaination. Had a look through the rest of Nuklear, and it looks like rather than using memmove directly, uses `NK_MEMCPY`, which could make sense here...
``` c
        /* memmove */
        char *dst = nk_ptr_add(char, s->buffer.memory.ptr, pos);
        char *src = nk_ptr_add(char, s->buffer.memory.ptr, pos + len);
        NK_MEMCPY(dst, src, s->buffer.allocated - (nk_size)(pos + len));
```

https://cplusplus.com/reference/cstring/memcpy/